metricbeat/module/kubernetes/container/data.go

```diff
@@ -127,17 +128,19 @@ func eventMapping(content []byte, perfMetrics *util.PerfMetricsCache, logger *lo
 				kubernetes2.ShouldPut(containerEvent, "memory.usage.node.pct", float64(container.Memory.UsageBytes)/nodeMem, logger)
 			}
 
-			cuid := util.ContainerUID(pod.PodRef.Namespace, pod.PodRef.Name, container.Name)
-			coresLimit := perfMetrics.ContainerCoresLimit.GetWithDefault(cuid, nodeCores)
-			memLimit := perfMetrics.ContainerMemLimit.GetWithDefault(cuid, nodeMem)
+			containerUID := util.ContainerUID(pod.PodRef.Namespace, pod.PodRef.Name, container.Name)
+			containerMetricsRepoId := util.GetMetricsRepoId(util.ContainerMetricSource, containerUID)
 
-			if coresLimit > 0 {
-				kubernetes2.ShouldPut(containerEvent, "cpu.usage.limit.pct", float64(container.CPU.UsageNanoCores)/1e9/coresLimit, logger)
+			containerCoresLimit := metricsRepo.GetWithDefault(containerMetricsRepoId, util.ContainerCoresLimitMetric, nodeCores)
+			containerMemLimit := metricsRepo.GetWithDefault(containerMetricsRepoId, util.ContainerMemoryLimitMetric, nodeMem)
+
+			if containerCoresLimit > 0 {
```

Comment:
I wonder why we don't check if `nodeCores > 0 && containerCoreLimits > nodeCores ` like we do in other metricsets 🤔

Reply:
I think this is the reasoning:
- in the metricset `container` a metric (eg. cpu) can either have a fixed limit or get the limit from the node. I assume that k8s will not allow a single container to set a limit greater than the node limits. I am not sure of this and I cannot find it anywhere in the docs. 
- in the metricset `node` in order to compute the pod limits we sum up all the limits for all the containers of that pod. If a container doesn't have a limit, we assign the node limit as default value. In case you have a multi-container pod with at least a container without limits, the sum of all limits can go over the node limits (because of node limits used a default value). Since that would be an error, in that case we can the limit to the node limit.

Reply:
Maybe it's worth writing this logic down so as our future-selfs not wondering about this next time? Maybe each field should mention in tis docs how it is calculated , based on what assumptions etc...Maybe not in this PR but in a follow-up :)